### PR TITLE
added search tests

### PR DESCRIPTION
**Pull request by AlexStefan17**
Implementation of https://github.com/adrianmaciuc/break-it-playwright-javascript-project/issues/17

Review changes are done.

### CODE REVIEW

File: tests/search/search.spec.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+test("Search in header for tees", async ({ page }) => {
+  await page.goto(homePageUrl);
+
+  // Enter "tee" and click on desired sugestion
+  await page
+    .getByPlaceholder(searchBar)
+    .pressSequentially("tee", { delay: 300 });
+
+  await page.locator(searchItems).getByText("tees").first().click();
+  await page.waitForTimeout(1000);
```
[the remark this answers is not in the document]
Comment (adrianmaciuc) [reply tests/search/search.spec.js:68]: nevermind, you do need this, its really tricky here. again this website is complicated

File: tests/search/search.spec.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+
+  await expect((await page.locator(products).all()).length).toBe(3);
+
+  await expect(await page.locator(pagination)).toHaveCount(0);
```
Comment (adrianmaciuc) [inline tests/search/search.spec.js:72]:
```suggestion
  await expect((await page.locator(products).all())).toHaveLength(3);

  await expect(await page.locator(pagination)).not.toBeVisible()
```

I see you didn't read my tip nr 8[ from my blog ](https://blog.martioli.com/playwright-tips-and-tricks-1/)

File: tests/search/search.spec.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+    sugestions.push(await sugestion.innerText());
+  }
+
+  await expect(sugestions).toContain("pants for women");
+
+  const sugestionAmount = await page
+    .locator(`#qs-option-${sugestions.indexOf("pants for women")}`)
+    .locator(amount)
+    .innerText();
+  await expect(sugestionAmount).toBe("25");
```
[the remark this answers is not in the document]
Comment (adrianmaciuc) [reply tests/search/search.spec.js:33]: You write it nice, but the only part that was a bit confusing for me was how hard it is to read the indexOf line and figure out what you are doing. Sometimes the builtin methods of playwright provide a more easy readable way.

File: tests/search/search.spec.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+const elementsList = "ul[role='listbox']";
+const searchItems = ".qs-option-name";
+const amount = ".amount";
+const products = ".product-item";
+const pagination = ".items.pages-items";
+
+test("Search in header for pants", async ({ page }) => {
+  await page.goto(homePageUrl);
+
+  // Enter "pants" and get sugestions
```
Comment (pmirel) [inline tests/search/search.spec.js:14]: you can remove the comments

File: tests/search/search.spec.js [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+    .getByPlaceholder(searchBar)
+    .pressSequentially("pants", { delay: 300 });
+  await expect(page.locator(elementsList)).toBeVisible();
+  const sugestionList = await page.locator(searchItems).all();
+
+  // Get sugestions
+  let sugestions = [];
+  for (let sugestion of sugestionList) {
+    sugestions.push(await sugestion.innerText());
+  }
```
Comment (pmirel) [inline tests/search/search.spec.js:25]: Is this necessary? all already returns a list of array. Can you make the assertion on that list?

Comment (pmirel) [reply tests/search/search.spec.js:25]: nevermind adi already wrote the code

**Review by pmirel — CHANGES_REQUESTED** [review]
other than what adi already said, remove the comments and add a link to the ticket in decription.

**Review by pmirel — APPROVED** [review]

File: tests/search/search.spec.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
...
+    })
+    .locator(amount)
+    .allInnerTexts();
+  await expect(optionPantsForWomen).toContain("25");
+});
+
+test("Search in header for tees for man", async ({ page }) => {
+  await page.goto(homePageUrl);
+
+  // Enter "tee" and click on desired sugestion
```
Comment (pmirel) [inline tests/search/search.spec.js:34]: remove comm

File: tests/search/search.spec.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+  await page
+    .getByPlaceholder(searchBar)
+    .pressSequentially("tee", { delay: 300 });
+
+  await page.locator(searchItems).getByText("tee for man").click();
+
+  await expect(await page.locator(products)).toHaveCount(12);
+
+  // Get number of pages
```
Comment (pmirel) [inline tests/search/search.spec.js:43]: remove comm

File: tests/search/search.spec.js [hunk 8]
```diff
@@ -0,0 +1,65 @@
...
+    .first()
+    .locator(".item .page")
+    .all();
+  await expect(pagesCount).toHaveLength(4);
+});
+
+test("Search in header for tees", async ({ page }) => {
+  await page.goto(homePageUrl);
+
+  // Enter "tee" and click on desired sugestion
```
Comment (pmirel) [inline tests/search/search.spec.js:55]: remove comm

File: tests/search/search.spec.js [hunk 9]
```diff
@@ -0,0 +1,65 @@
...
+  await page
+    .getByPlaceholder(searchBar)
+    .pressSequentially("tee", { delay: 300 });
+
+  await page.locator(searchItems).getByText("tees").first().click();
+  await page.waitForTimeout(1000);
+
+  await expect(await page.locator(products).all()).toHaveLength(3);
+  await expect(await page.locator(pagination)).not.toBeVisible(0);
+});
```
Comment (pmirel) [inline tests/search/search.spec.js:65]: you acctualy try to use .toBeHidden() 
And remove 0 from argument

**Review by pmirel — CHANGES_REQUESTED** [review]

**Review by pmirel — APPROVED** [review]

**Review by adrianmaciuc — APPROVED** [review]

**Pull request merged**